pkg/workloadapi/workload.proto

```diff
@@ -81,6 +79,20 @@ message Service {
   // Note: this applies only to connecting directly to the workload; when waypoints are used, the waypoint's load_balancing
   // configuration is used.
   LoadBalancing load_balancing = 8;
+
+  // IP families provides configuration about the IP families this service supports.
+  IPFamilies ip_families = 9;
```

Comment:
Is this really  needed, can infer from `addresses`

Reply:
unfortunately for headless services there is no vip. If it wasn't for that I wouldn't have added it probably . 

see https://github.com/istio/ztunnel/issues/1152

Reply:
How is headless processed in ztunnel now, it has no ip associated on service. 

It should be matched by a workload ip. Since workload has the ip, can we infer from it. Anyway, i am not sure how you want ztunnel to do with dual stack

Reply:
It is for DNS. The linked ztunnel PR has the ztunnel implementation and tests